### ISSUE TITLE
invalid 'id' type in frontmatter caused by invalid Obsidian property

### ISSUE THREAD

**Issue by one2three4f** [epwalsh/obsidian.nvim#209]
### 🐛 Describe the bug

New Obsidian update supports properties for files now, but causes errors when the property values type isnt a strings, so types date, date and time, and checkboxes,  aren't supported. 

Printing out the value and types, I get the dates are a number and the checkboxes are a boolean.

To reproduce:
open any .md file in Obsidian
create property
select 'id'
choose a date propert type (any other than text, so list, numbers, checkboxes, ...)
input a date on obsidian and save
results in invalid id type error

from notes.lua line 221
``` lua 
-- Parse the frontmatter YAML.
  local metadata = nil
  if #frontmatter_lines > 0 then
    local frontmatter = table.concat(frontmatter_lines, "\n")
    local ok, data = pcall(yaml.loads, frontmatter)
    if type(data) == "string" then
      data = {}
    end
    if ok then
      ---@diagnostic disable-next-line: param-type-mismatch
      for k, v in pairs(data) do
        if k == "id" then
          if type(v) == "string" then
            id = v
          else
            echo.warn("Invalid 'id' in frontmatter for " .. tostring(path))
          end
```

not sure if intended since empty tags dont crate an error but empty aliases create this error:
`
[Obsidian] Invalid 'aliases' in frontmatter for PATH
`

My temporary solution is to add the new property types in line 218 in notes.lua but on every save the new property types goes back to the original string.

``` lua
        if k == "id" or type(v) == "boolean" or type(v) == "number" then

```

### Versions

```
nvim --version
NVIM v0.10.0-dev-1055+gb7734c4ec
Build type: RelWithDebInfo
LuaJIT 2.1.1693350652
Run "nvim -V1 -v" for more info
```
```
nvim --headless -c 'lua print("Obsidian v" .. require("obsidian").VERSION)' -c q
Obsidian v1.14.2
```

**Comment by one2three4f** [comment]
Love the extension btw

**Comment by epwalsh** [comment]
Hey @one2three4f thanks for bringing this up. There was a couple bugs in the YAML parsing code which are now fixed (96ac0d9). I also changed it so `id` can be a string or a number. Do you have a use-case where `id` would be a boolean?

**Comment by one2three4f** [comment]
Personally no, but the new Obisidan property types support checkboxes which are boolean values.
